submissions/Oleksii_Drohachov/Memory_pair_game/script.js

```diff
@@ -0,0 +1,220 @@
+  scoreCount.innerHTML = "";
+  gameboard.innerHTML = "";
+  modal.innerHTML = "";
+  modal.classList.remove("shown");
+  prepareGame();
+}
+
+function prepareGame() {
+  const cardsArray = [
+    {
+      name: "beast",
+      path: "img/Beast.jpg",
+    },
+    {
+      name: "beast",
+      path: "img/Beast.jpg",
+    },
+    {
+      name: "death",
+      path: "img/Death.jpg",
+    },
+    {
+      name: "death",
+      path: "img/Death.jpg",
+    },
+    {
+      name: "fire",
+      path: "img/Fire.jpg",
+    },
+    {
+      name: "fire",
+      path: "img/Fire.jpg",
+    },
+    {
+      name: "life",
+      path: "img/Life.jpg",
+    },
+    {
+      name: "life",
+      path: "img/Life.jpg",
+    },
+    {
+      name: "light",
+      path: "img/Light.jpg",
+    },
+    {
+      name: "light",
+      path: "img/Light.jpg",
+    },
+    {
+      name: "metal",
+      path: "img/Metal.jpg",
+    },
+    {
+      name: "metal",
+      path: "img/Metal.jpg",
+    },
+  ];
+  let shuffled = [...cardsArray];
+  shuffled.sort(function () {
+    return 0.5 - Math.random();
+  });
+  let score = 0;
+  let guessedCardPairs = 0;
+
+  renderScoreAndReset();
+  renderCards(shuffled);
+  cardsHandler();
+
+  const scoreSpan = document.querySelector(".score__span");
+
+  function renderScoreAndReset() {
+    const scoreSpan = document.createElement("span");
+    scoreSpan.classList.add("score__span");
+    scoreSpan.textContent = `Number of tries: ${score}`;
+    const resetBtn = document.createElement("button");
+    resetBtn.addEventListener("click", renderModalForReset);
+    resetBtn.classList.add("reset__btn");
+    resetBtn.textContent = "RESTART";
+    scoreCount.append(scoreSpan);
+    scoreCount.append(resetBtn);
+  }
+
+  function updateScore() {
+    scoreSpan.textContent = `Number of tries: ${score}`;
+  }
+
+  function renderCards(arr) {
+    const cardWrapper = document.createElement("div");
+    cardWrapper.classList.add("card__wrapper");
+
+    arr.forEach((item) => {
+      const card = document.createElement("div");
+      card.classList.add("card");
+      card.setAttribute("data-name", item.name);
+      const frontPage = document.createElement("img");
+      frontPage.classList.add("front__page");
+      frontPage.setAttribute("src", item.path);
+      const backPage = document.createElement("img");
+      backPage.classList.add("back__page");
+      backPage.setAttribute("src", "img/backside.jpeg");
+
+      card.append(frontPage);
+      card.append(backPage);
+      cardWrapper.append(card);
+    });
+    gameboard.append(cardWrapper);
+  }
+
+  function cardsHandler() {
+    const cards = document.querySelectorAll(".card");
+    cards.forEach((card) => card.addEventListener("click", flipCard));
+
+    let hasFlipped = false;
+    let lockBoard = false;
+    let firstCard, secondCard;
+
+    function flipCard() {
+      if (lockBoard) return;
+      if (this === firstCard) return;
+
+      this.classList.add("flip");
```

Comment:
You can easily lose context and the corresponding value of `this` pass selected card as a parameter for the function.